packages/core/tests/integration/components/navi-visualizations/goal-gauge-test.ts

```diff
@@ -0,0 +1,364 @@
+  @model={{this.model}}
+  @options={{this.options}}
+/>`;
+
+type ComponentArgs = GoalGaugeVisualization['args'];
+
+interface TestContext extends Context, ComponentArgs {
+  request: RequestFragment;
+}
+
+module('Integration | Component | navi-visualization/goal gauge ', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function(this: TestContext) {
+    const store = this.owner.lookup('service:store') as StoreService;
+    this.options = { metricCid: 'cid_pageViews', baselineValue: 290000000, goalValue: 310000000 };
+    this.request = store.createFragment('bard-request-v2/request', {
+      table: null,
+      columns: [
+        {
+          cid: 'cid_pageViews',
+          type: 'metric',
+          field: 'pageViews',
+          parameters: {},
+          alias: 'Page Views',
+          source: 'bardOne'
+        }
+      ],
+      filters: [],
+      sorts: [],
+      limit: null,
+      dataSource: 'bardOne',
+      requestVersion: '2.0'
+    });
+    this.set(
+      'model',
+      arr([
+        { request: this.request, response: NaviFactResponse.create({ rows: [{ pageViews: 3030000000 }], meta: {} }) }
+      ])
+    );
+  });
+
+  test('goal-gauge renders correctly', async function(this: TestContext, assert) {
+    assert.expect(6);
+
+    await render(TEMPLATE);
+
+    assert.dom('.c3-chart-component svg').exists('gauge component renders');
+
+    assert.dom('.value-title').hasText('3.03B', 'value title is correctly displayed');
+
+    assert.dom('.metric-title').hasText('Page Views', 'the default metric title is correctly displayed');
+
+    assert.dom('.c3-chart-arcs-gauge-min').hasText('290M', 'min gauge label is correctly displayed');
+
+    assert.dom('.c3-chart-arcs-gauge-max').hasText('310M', 'max gauge label is correctly displayed');
+
+    assert.dom('.goal-title').hasText('310M Goal', 'goal title is correctly displayed');
+  });
+
+  test('goal-gauge renders correctly with multi datasource', async function(this: TestContext, assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store') as StoreService;
+    this.request = store.createFragment('bard-request-v2/request', {
+      table: null,
+      columns: [
+        {
+          cid: 'cid_available',
+          type: 'metric',
+          field: 'available',
+          parameters: {},
+          alias: 'How many are available',
+          source: 'bardTwo'
+        }
+      ],
+      filters: [],
+      sorts: [],
+      limit: null,
+      dataSource: 'bardOne',
+      requestVersion: '2.0'
+    });
+
+    this.set(
+      'model',
+      arr([{ request: this.request, response: NaviFactResponse.create({ rows: [{ available: 3030000000 }] }) }])
+    );
+    this.set('options', { metricCid: 'cid_available', baselineValue: 290000000, goalValue: 310000000 });
+
+    await render(TEMPLATE);
+
+    assert.dom('.metric-title').hasText('How many are available', 'the default metric title is correctly displayed');
```

Comment:
We might want to test multi dataSource without an alias first and then with alias